numpy/linalg/tests/test_linalg.py

```diff
@@ -320,43 +348,87 @@ def _check_cases(func, cases):
             raise AssertionError(msg)
 
 
-class LinalgTestCase(object):
+class LinalgSquareTestCase(object):
 
     def test_sq_cases(self):
-        _check_cases(self.do, SQUARE_CASES)
+        _check_cases(self.do,
+            require=CaseFlags.square,
+            exclude=CaseFlags.generalized | CaseFlags.empty)
+
+    def test_empty_sq_cases(self):
+        _check_cases(self.do,
+            require=CaseFlags.square | CaseFlags.empty,
+            exclude=CaseFlags.generalized)
 
 
 class LinalgNonsquareTestCase(object):
 
-    def test_sq_cases(self):
-        _check_cases(self.do, NONSQUARE_CASES)
+    def test_nonsq_cases(self):
+        _check_cases(self.do,
+            require=CaseFlags.none,
+            exclude=CaseFlags.generalized | CaseFlags.square | CaseFlags.hermitian | CaseFlags.empty)
```

Comment:
See above. I think these would all be clearer as a list of strings and take up less room to boot.